packages/java-parser/src/productions/classes.js

```diff
@@ -606,9 +606,14 @@ function defineRules($, t) {
   // https://docs.oracle.com/javase/specs/jls/se11/html/jls-8.html#jls-EnumConstantList
   $.RULE("enumConstantList", () => {
     $.SUBRULE($.enumConstant);
-    $.MANY(() => {
-      $.CONSUME(t.Comma);
-      $.SUBRULE2($.enumConstant);
+    $.MANY({
+      GATE: () =>
+        $.LA(1).tokenType === t.Comma &&
+        ($.LA(2).tokenType === t.Identifier || $.LA(2).tokenType === t.At),
```

Comment:
comparing a tokenType directly to an Identifier is unsafe and would miss some edge cases.
That is because Identifier is used as a Token Category.
- https://sap.github.io/chevrotain/docs/features/token_categories.html
- Fix: use tokenMatcher: https://github.com/jhipster/prettier-java/blob/1a4f6525ce2b86be36459e7e8db5252d94932d39/packages/java-parser/src/productions/interfaces.js#L372
- https://github.com/jhipster/prettier-java/blob/1a4f6525ce2b86be36459e7e8db5252d94932d39/packages/java-parser/src/tokens.js#L177-L179